UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs

```diff
@@ -9,56 +9,51 @@ namespace Testing.Common.ActiveDirectory
 {
     public static class ActiveDirectoryUser
     {
+        private static string ApiBaseUrl => $"https://graph.microsoft.com/v1.0/{TestConfig.Instance.AzureAd.TenantId}";
 
         public static async Task<bool> IsUserInAGroup(string user, string groupName, string token)
         {
-            using (var client = new HttpClient())
-            {
-                client.DefaultRequestHeaders.Authorization =
-                    new AuthenticationHeaderValue("Bearer", token);
-                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get,
-                    $@"https://graph.microsoft.com/v1.0/users/{user}/memberOf");
-                var result = client.SendAsync(httpRequestMessage).Result;
-                var content = await result.Content.ReadAsStringAsync();
-                return content.Contains(groupName);
-            }
+            var url = $@"{ApiBaseUrl}/users/{user}/memberOf";
+            var groupsResult = await SendGraphApiRequest(url, HttpMethod.Get, token);
+            return groupsResult.Contains(groupName);
         }
 
-        public static bool RemoveTheUserFromTheGroup(string user, string groupId, string token)
+        public static async Task RemoveTheUserFromTheGroup(string user, string groupId, string token)
         {
-            using (var client = new HttpClient())
-            {
-                client.DefaultRequestHeaders.Authorization =
-                    new AuthenticationHeaderValue("Bearer", token);
-                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Delete,
-                    $@"https://graph.microsoft.com/v1.0/groups/{groupId}/members/{user}/$ref");
-                var result = client.SendAsync(httpRequestMessage).Result;
-                return result.IsSuccessStatusCode;
-            }
+            var url = $@"{ApiBaseUrl}/groups/{groupId}/members/{user}/$ref";
+            await SendGraphApiRequest(url, HttpMethod.Delete, token);
+            Console.WriteLine($"Deleted group '{groupId}' from user: {user}");
         }
 
-        public static async Task<bool> DeleteTheUserFromAd(string user, string token)
+        public static async Task DeleteTheUserFromAd(string user, string token)
+        {
+            var url = $@"{ApiBaseUrl}/users/{user}";
+            await SendGraphApiRequest(url, HttpMethod.Delete, token);
+            Console.WriteLine($"Deleted user: {user}");
+        }
+
+        private static async Task<string> SendGraphApiRequest(string url, HttpMethod method, string token)
```

Comment:
Maybe create an object to capture response statuscode and message, then you can on line 32 say something like
object.statuscode == OK ? log("deleted nicely" ) : log("not deleted mate, status {object.status}")

Reply:
As we talked about, I don't really care about the response code. Basically, if it returns it will have passed with a successful response